scripts/policy/protocols/conn/remove-ip-proto-log.zeek

```diff
@@ -0,0 +1,17 @@
+#! This script filters the ip_proto field out of the conn.log.
+
+@load base/protocols/conn
+
+module Conn;
+
+event zeek_init() &priority=-100
+	{
+	local filter = Log::get_filter(Conn::LOG, "default");
```

Comment:
We could also do the following:

```
redef Conn::Info$ip_proto -= { &log }
```

This would hide `ip_proto` from any logic that computes/creates conn.log schmeas based on the presence of `&log`. Also works for all filters rather than just the default filter.

Should we also  disable tracking of unknown/unsupported protocols here. If ip_proto isn't logged, maybe we shouldn't even log and track such connections, so they are fully excluded from conn.log?

```
redef PacketAnalyzer::IP::default_analyzer = NONE;  # or whatever works
# or maybe a BifConst that is checked in the unknown ip analyzer instead.
redef PacketAnalyzer::IP::track_unknown|unsupported_connections = F;
```

Reply:
Yes please, to both — at last a use-case for that `&log` redef! :-) This is also nicer than iterating over all currently defined filters.

The effect should be as if that whole feature were not there, which means we should not just suppress the column, but also avoid the additional connections. I initially thought of just filtering via a policy hook, but disabling the analyzer (or part of it) would be much nicer.

Reply:
I assume this should also disable the protocol name policy if someone opted to load both policies?

Reply:
I don't really like the idea of setting it to `NONE` because someone else might have redef'd it already, and our setting it to `NONE` would break that. I'll opt for the variable but that still means we're calling an analyzer to do no work.

Re: loading both policies, given there's no way to conditionally redef a record at startup, I think we just live with the fact that someone can do that. If they load the names policy and the removal policy, they won't get connections with unknown protocols in the log but they'll still get the `ip_proto_name` field that should hopefully match the `proto` field.

Reply:
> I decided that this PR is the one that sets it to something other than NONE, so setting it back to that is fine.

There's actually no good way to unset a default analyzer. Since we're passing a Tag, there isn't something you can pass that indicates "ignore this". A "none" tag would have to be added somewhere. I opted for just calling `disable_analyzer` on the unknown IP analyzer instead.

> 
> Re: loading both policies, given there's no way to conditionally redef a record at startup, I think we just live with the fact that someone can do that. If they load the names policy and the removal policy, they won't get connections with unknown protocols in the log but they'll still get the `ip_proto_name` field that should hopefully match the `proto` field.

I added some extra docs to the protocol strings policy script about this. I'm also considering renaming that script to `ip_proto_strings.zeek` since the current name isn't descriptive enough.